src/utils/desktop.js

```diff
@@ -147,9 +148,13 @@ export const resourceResolver = (core) => {
 
   const soundsEnabled = () => !!getSoundThemeName();
 
-  const icon = path => {
+  const icon = (name) => {
     const theme = getThemeName('icons');
-    return core.url(`icons/${theme}/icons/${path}`); // FIXME: Use metadata ?
+    const extension = core.make('osjs/packages')
```

Comment:
A few things here:

1. Probably better to hoist the service provider destructor to the top of the scope:
```js
export const resourceResolver = (core) => {
  const {getMetadataFromName} = core.make('osjs/packages');
```
2. This does not cover cases of invalid themes, nor if if the `icons` property is missing (make sure to create tests for this)
```js
const metadata = getMetadataFromName(theme)
const iconDefinitions = metadata ? metadata.icons || {} : {}
const extension = iconDefinitions[name] || 'png'
```

(I'm not sure if the current Babel setup takes care of optionals... above could be written as `iconDefinitions = metadata?.icons || {}` if that was the case. Gonna look into that).